### PR TITLE
feat(android): new event for empty TextFields

### PR DESCRIPTION
**Pull request by m1ga**
The `change` event on a TextField is only called when there is content in the field. If it is empty and you press "backspace" again you won't get any event. 

I've added an event: `empty` that will fire. 

```js
const win = Ti.UI.createWindow({});
var tf1 = Ti.UI.createTextField({
	width: 150,
	left: 10
});
var tf2 = Ti.UI.createTextField({
	width: 150,
	right: 10,
	value: "remove me"
});
tf2.addEventListener("empty", function(e) {
	console.log("key", e.keyCode)
	tf1.focus();
})
tf2.addEventListener("change", function(e) {
	console.log("change", tf2.value.length)
})
win.add(tf1);
win.add(tf2);
win.addEventListener("open", function() {
	tf2.focus();
})
win.open();
```

https://github.com/tidev/titanium-sdk/assets/4334997/f10a081e-8810-4744-a685-d1d4f67d9d9e

TODO:
* [x] Docs
* [ ] rename it?
* [ ] someone needs to check if iOS has something similar

### CODE REVIEW

**Comment by jordanbisato** [comment]
Isn't necessary to check if the TextField is empty before trigger the emptyField event?

**Comment by m1ga** [comment]
as far as I can tell from testing it will only fire when the field is empty, so no text is in there and you press a key (See video above). You have to check the keyCode in Ti to see if it is the back key. I can add a check for the length but currently it looks like it is not necessary unless you have an example where it breaks

File: apidoc/Titanium/UI/TextField.yml [hunk 1]
```diff
@@ -610,6 +610,15 @@ events:
         summary: New value of the field.
         type: String
 
+  - name: emptyField
```
Comment (hansemannn) [inline apidoc/Titanium/UI/TextField.yml:613]: Rename to `empty`?

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIText.java [hunk 2]
```diff
@@ -135,6 +135,20 @@ public void onLayoutChange(
 		this.tv.setOnEditorActionListener(this);
 		this.tv.setOnFocusChangeListener(this);
 		this.tv.setIncludeFontPadding(true);
+		this.tv.setOnKeyListener(new View.OnKeyListener()
```
Comment (hansemannn) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIText.java:138]: Only add the listener if the event is listening for it

**Review by hansemannn — CHANGES_REQUESTED** [review]

**Comment by m1ga** [comment]
Fixed both comments

**Review by hansemannn — APPROVED** [review]

**Pull request merged**